python/ray/data/dataset.py

```diff
@@ -613,6 +613,55 @@ def do_shuffle(block_list, clear_input_blocks: bool, block_udf, remote_args):
         )
         return Dataset(plan, self._epoch, self._lazy)
 
+    def random_sample(
+        self, fraction: float, *, seed: Optional[int] = None
+    ) -> "Dataset[T]":
+        """Randomly samples a fraction of the elements of this dataset.
+
+        Note that the exact number of elements returned is not guaranteed.
```

Comment:
While this is true, we can make it clear that the expected number of elements is fraction*total rows.

Reply:
Noted in 17e8e5c